### ISSUE TITLE
GPT PMBR size mismatch with second disk

### ISSUE THREAD

**Issue by yamatoRT** [vagrant-libvirt/vagrant-libvirt#1308]
Hi all,

when I use a second disk in box, the VM wrong detect the disk. When I use `fdisk`, I have this message :
```
GPT PMBR size mismatch (20971528 != 20971519) will be corrected by w(rite).
The backup GPT table is corrupt, but the primary appears OK, so that will be used.

Disk /dev/sdb: 10 GiB, 10737418240 bytes, 20971520 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 3FF63104-B751-AB4A-A567-96BBCD052F77

Device     Start      End  Sectors Size Type
/dev/sdb1   2048 20971495 20969448  10G Linux filesystem
```

### Steps to reproduce
1. Create alias  :
```bash
alias vagrant='
  mkdir -p ~/.vagrant.d/{boxes,data,tmp}; \
  docker run -it --rm \
    -e LIBVIRT_DEFAULT_URI \
    -v /var/run/libvirt/:/var/run/libvirt/ \
    -v ~/.vagrant.d:/.vagrant.d \
    -v $(pwd):$(pwd) \
    -w $(pwd) \
    --network host \
    vagrantlibvirt/vagrant-libvirt:latest \
    vagrant'
```
2. Set a attached Vagrantfile in current directory
3. vagrant up
4. Into the vm, run the command `sudo fdisk -l`

### Expected behaviour
The second disk is normally mounted

### Actual behaviour
The size of disk doesn't match with the GPT PMBR size 

### System configuration
**OS/Distro version:**:
Linux archbox 4.3.3-3-ARCH

**Libvirt version:**
```
vagrant-libvirt (0.5.2, global)
  - Version Constraint: 0.5.2
```
**Output of `vagrant version; vagrant plugin list`:**
```
Installed Version: 2.2.10
Latest Version: 2.2.16
 
To upgrade to the latest version, visit the downloads page and
download and install the latest version of Vagrant from the URL
below:

  https://www.vagrantup.com/downloads.html

If you're curious what changed in the latest release, view the
CHANGELOG below:

  https://github.com/hashicorp/vagrant/blob/v2.2.16/CHANGELOG.md
Starting with UID: 1000, GID: 1000
vagrant-libvirt (0.5.2, global)
  - Version Constraint: 0.5.2
```

**Attach Output of `VAGRANT_LOG=debug vagrant ... --provider=libvirt >vagrant.log 2>&1`**
[vagrant.log](https://github.com/vagrant-libvirt/vagrant-libvirt/files/6625747/vagrant.log)

**A Vagrantfile to reproduce the issue:**
```
Vagrant.configure("2") do |config|
  config.vm.box = "yamatoRT/arch-linux"
  config.vm.box_version = "0.0.2"
  config.nfs.verify_installed = false
  config.nfs.functional = false
  config.vm.synced_folder "." "/vagrant", disabled:true
end
```

**Are you using upstream vagrant package or your distros package?**
Upstream

**Comment**

I think the problem is into [handle_box_image.rb](https://github.com/vagrant-libvirt/vagrant-libvirt/blob/master/lib/vagrant-libvirt/action/handle_box_image.rb) with the output of `qemu-img`.
```
> qemu-img info ~/.vagrant.d/boxes/yamatoRT-VAGRANTSLASH-arch-linux/0.0.1/libvirt/disk2.qcow2                                                                                                    
image: ~/.vagrant.d/boxes/yamatoRT-VAGRANTSLASH-arch-linux/0.0.1/libvirt/disk2.qcow2
file format: qcow2
virtual size: 10 GiB (10737423360 bytes)
disk size: 640 KiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    compression type: zlib
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
    extended l2: false
```
The virtual size is **10G** but **10737423360/(1024^3) = 10.00000476**, normally the virtual size must be **11G**

**Comment by electrofelix** [comment]
That does indeed look to be the issue, I suspect we need to normalize to storing the byte size internally, default to assuming the provided size is in gigabytes, but otherwise start accepting size qualifiers as part of the virtual_size where specified and finally pretty print the summary size as well as full size in bytes when displaying.

**Comment by electrofelix** [comment]
@yamatoRT going to assume that the bug is fixed now based on it being released as part of v0.5.3.
